lb.tf

```diff
@@ -0,0 +1,55 @@
+resource "aws_lb" "main" {
+  count = var.nlb_enabled ? 1 : 0
+
+  name               = var.nlb_name
+  internal           = var.nlb_internal
+  load_balancer_type = "network"
+  subnets            = var.nlb_subnet_ids
+
+  enable_cross_zone_load_balancing = var.enable_cross_zone_load_balancing
+  enable_deletion_protection       = var.enable_deletion_protection
+
+  tags = var.tags
+}
+
+resource "aws_lb_target_group" "main" {
+  count = var.nlb_enabled ? 1 : 0
+
+  name        = var.nlb_name
+  port        = var.nlb_tg_port
+  protocol    = var.nlb_tg_protocol
+  target_type = "ip"
+  vpc_id      = var.vpc_id
```

Comment:
Instead of introducing another variable, can we fetch vpc_id based on subnet_id? Through data source.

Reply:
Amazon API doesn't have such filter